include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -35,7 +35,7 @@ using void_type = typename ::std::enable_if<::std::is_void<_Tp>::value, _Tp>::ty
 template <typename _Tp>
 using non_void_type = typename ::std::enable_if<!::std::is_void<_Tp>::value, _Tp>::type;
 
-#if _USE_GROUP_ALGOS
+#if _USE_GROUP_ALGOS && defined(SYCL_LANGUAGE_VERSION) && defined (__INTEL_LLVM_COMPILER)
```

Comment:
Using only `__INTEL_LLVM_COMPILER` should be enough. This header is supposed to be included if a compiler supports SYCL. 
Let's leave a comment describing the limitation because it is not obvious.
Do not forget to update lines like `#else //_USE_GROUP_ALGOS` in accordance with the condition.

Reply:
Done.